internal/common/pid.go

```diff
@@ -8,34 +8,32 @@ import (
 	"github.com/edulinq/autograder/internal/util"
 )
 
-func CheckAndHandlePid() bool {
+func checkAndHandleStalePid() bool {
```

Comment:
I had it as logging an error because you made the suggestion earlier to log it and return a bool. If we're now going to be returning an error, would saying "another instance of the autograder is already running" be a valid error on line 37? I'm only asking because you mentioned we shouldn't say that and just let the client deal with it.

Reply:
> I'm not a fan of most the logs happening in here.

That's why I said "most" of the logging.
The warn is fine, but we don't want to over-generalize.

I believe you need the two returns, because there are three cases here:
1. An error happened (err != nil).
2. Another instance of the server is up. (false, nil)
3. No other server is up, we are good to go. (true, nil)